File: src/Entity/Actualite.php
<?php

namespace App\Entity;

use App\Repository\ActualiteRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ActualiteRepository::class)]
class Actualite
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id_actualite= null;

    #[ORM\Column(length: 50)]
    private ?string $Titre = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $Description = null;

    #[ORM\Column(length: 55)]
    private ?string $Type_pub_cible = null;

    #[ORM\Column(length: 255)]
    private ?string $Theme = null;

    #[ORM\OneToOne(inversedBy: 'actualite', cascade: ['persist', 'remove'])]
    #[ORM\JoinColumn(name: 'evenement_id', referencedColumnName: 'id_evenement')]
    private ?Evenement $Evenement = null;

    #[ORM\ManyToOne(inversedBy: 'actualites')]
    private ?User1 $User = null;

    public function getId(): ?int
    {
        return $this->id_actualite;
    }

    public function getTitre(): ?string
    {
        return $this->Titre;
    }

    public function setTitre(string $Titre): static
    {
        $this->Titre = $Titre;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->Description;
    }

    public function setDescription(?string $Description): static
    {
        $this->Description = $Description;

        return $this;
    }

    public function getTypePubCible(): ?string
    {
        return $this->Type_pub_cible;
    }

    public function setTypePubCible(string $Type_pub_cible): static
    {
        $this->Type_pub_cible = $Type_pub_cible;

        return $this;
    }

    public function getTheme(): ?string
    {
        return $this->Theme;
    }

    public function setTheme(string $Theme): static
    {
        $this->Theme = $Theme;

        return $this;
    }

    public function getEvenement(): ?Evenement
    {
        return $this->Evenement;
    }

    public function setEvenement(?Evenement $Evenement): static
    {
        $this->Evenement = $Evenement;

        return $this;
    }

    public function getUser(): ?User1
    {
        return $this->User;
    }

    public function setUser(?User1 $User): static
    {
        $this->User = $User;

        return $this;
    }

}

File: src/main/java/Entity/Actualite.java
package Entity;

public class Actualite {
    private int id_actualite;
    private String Titre ;
    private String Description;
    private String Type_pub_cible;
    private String Theme;

    private int id_user;


    public Actualite(String titre, String description, String type_pub_cible, String theme, int id_user) {
        Titre = titre;
        Description = description;
        Type_pub_cible = type_pub_cible;
        Theme = theme;
        this.id_user = id_user;
    }

    public int getId_user() {
        return id_user;
    }

    public Actualite(int id_actualite, String titre, String description, String type_pub_cible, String theme) {
        this.id_actualite = id_actualite;
        this.Titre = titre;
        this.Description = description;
        this.Type_pub_cible = type_pub_cible;
        this.Theme = theme;
    }

    public Actualite(String titre, String description, String type_pub_cible, String theme) {
        this.Titre = titre;
        this.Description = description;
        this.Type_pub_cible = type_pub_cible;
        this.Theme = theme;
    }

    public int getId_actualite() {
        return id_actualite;
    }

    public void setId_actualite(int id_actualite) {
        this.id_actualite = id_actualite;
    }

    public String getTitre() {
        return Titre;
    }

    public void setTitre(String titre) {
        this.Titre = titre;
    }

    public String getDescription() {
        return Description;
    }

    public void setDescription(String description) {
        this.Description = description;
    }

    public String getType_pub_cible() {
        return Type_pub_cible;
    }

    public void setType_pub_cible(String type_pub_cible) {
        this.Type_pub_cible = type_pub_cible;
    }

    public String getTheme() {
        return Theme;
    }



    public void setTheme(String theme) {
        this.Theme = theme;
    }

    @Override
    public String toString() {
        return "Actualite{" +
                "id_actualite=" + id_actualite +
                ", Titre='" + Titre + '\'' +
                ", Description='" + Description + '\'' +
                ", Type_pub_cible='" + Type_pub_cible + '\'' +
                ", Theme='" + Theme + '\'' +
                '}';
    }
}
